Dockerfile

```diff
@@ -1,8 +1,16 @@
-FROM golang:1.8.5-jessie
+FROM golang:alpine AS build
 # create a working directory
 WORKDIR /go/src/app
+#COPY go.mod go.sum ./
+#RUN go mod download
 # add source code
 COPY . .
 # run main.go
-RUN go run main.go
-#CMD ["bash"]
\ No newline at end of file
+RUN go build -o main
+#CMD ["go", "run", "main.go"]
+
+FROM scratch
+
+WORKDIR /go/src/app
+COPY --from=build /go/src/app/main .
+CMD ["./main"]
```

Comment:
This is not really a request or anything, just a sidenote that since main expects command line arguments, this line actually does nothing, but the fact that it runs is enough for me, since this will change soon enough anyway.